core/src/lib.rs

```diff
@@ -102,3 +103,9 @@ impl<F: Fn(Pin<Box<dyn Future<Output = ()> + Send>>)> Executor for F {
         self(f)
     }
 }
+
+impl Executor for ThreadPool {
+    fn exec(&self, future: Pin<Box<dyn Future<Output = ()> + Send>>) {
+        self.spawn_ok(future)
+    }
+}
```

Comment:
I am leaning towards removing this in a future version. I don't see why we need it.

It can be deprecated in a follow-up though.

I'll open an issue.

Reply:
I'm not quite sure what you mean. Removing ThreadPool?

Reply:
Removing the possibility of using `ThreadPool` for the connection tasks. Any async Rust program will use some kind of executor so we can just spawn the tasks into those worker threads instead of spinning up some threads on our own.

Reply:
Yes, you are right. But let's do that separately.

Reply:
Done: https://github.com/libp2p/rust-libp2p/issues/3107